### PR TITLE
3110 zh disallow unknown extensions

### PR DESCRIPTION
**Pull request by lu-wenhua**
-Added allowKnownExtensionsOnly() method which serves to exclude all unknown extensions when called. 
-Added building test for allowKnownExtensionsOnly() to make sure the program can build 
-Added allowKnownExtensionsOnly() documentation.

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/interceptor/validation/RepositoryValidatingRuleBuilder.java [hunk 1]
```diff
@@ -305,6 +305,15 @@ public FinalizedRequireValidationRule allowAnyExtensions() {
 				return this;
 			}
 
+			/**
+			 * Configure the validator to reject unknown extensions
+			 */
+			@Nonnull
+			public FinalizedRequireValidationRule allowKnownExtensionsOnly() {
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/interceptor/validation/RepositoryValidatingRuleBuilder.java:312]: I recommend renaming to:
`rejectUnknownExtensions()`

**Review by fil512 — CHANGES_REQUESTED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/interceptor/validation/RepositoryValidatingRuleBuilder.java (hunk 1 above)
Comment (lu-wenhua) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/interceptor/validation/RepositoryValidatingRuleBuilder.java]: completed as of now!

File: hapi-fhir-docs/src/main/java/ca/uhn/hapi/fhir/docs/RepositoryValidatingInterceptorExamples.java [hunk 2]
```diff
@@ -121,8 +120,10 @@ public void requireValidationToDeclaredProfilesAdditionalOptions() {
 			.forResourcesOfType("Patient")
 			.requireValidationToDeclaredProfiles()
 
-			// Configure the validator to never reject extensions
-			.allowAnyExtensions()
+			// Configure the validator to reject unknown extensions
+			// by default, all extensions are accepted and to undo this rejection
+			// call allowAnyExtensions()
```
Comment (fil512) [inline hapi-fhir-docs/src/main/java/ca/uhn/hapi/fhir/docs/RepositoryValidatingInterceptorExamples.java:125]: does this method still exist?

**Review by fil512 — APPROVED** [review]

File: hapi-fhir-docs/src/main/java/ca/uhn/hapi/fhir/docs/RepositoryValidatingInterceptorExamples.java (hunk 2 above)
Comment (lu-wenhua) [reply hapi-fhir-docs/src/main/java/ca/uhn/hapi/fhir/docs/RepositoryValidatingInterceptorExamples.java:125]: the method still exists.

**Pull request merged**